File: test/matlab/test_nabla_operator.m
clear all;


addpath('../../lib/matlab/');

x_rand = randn(64, 64, 32);
y_rand = randn(64, 64, 32, 3);

Ax = mex_nabla3_operator(x_rand);

ATy = mex_nabla3_operator_adjoint(y_rand);

lhs = sum(y_rand(:).*Ax(:))
rhs = sum(ATy(:).*x_rand(:))
abs(lhs - rhs)

File: test/python/test_nabla_operator.py
import sys
sys.path.insert(0, '../../lib/python')

import PyNablaOperator

import numpy as np

nabla_op = PyNablaOperator.NablaOperator()

x_rand = np.random.randn(64, 64).astype(np.float32)
y_rand = np.random.randn(2, 64, 64).astype(np.float32)

Ax = nabla_op.forward(x_rand)
ATy = nabla_op.adjoint(y_rand)
lhs = np.sum(Ax * y_rand)
rhs = np.sum(x_rand * ATy)
print(lhs, rhs, np.abs(lhs - rhs))
